keps/3125-maximum-execution-time/README.md

```diff
@@ -0,0 +1,207 @@
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [API](#api)
+    - [Workload Spec](#workload-spec)
+    - [Workload Status](#workload-status)
+  - [Constants](#constants)
+  - [Controller](#controller)
+    - [Workload](#workload)
+    - [Jobs / Jobframework](#jobs--jobframework)
+  - [Webhooks](#webhooks)
+    - [Workload](#workload-1)
+    - [Jobs](#jobs)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+      - [controller/core/workload](#controllercoreworkload)
+      - [controller/jobs/job](#controllerjobsjob)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+Add the ability to set a maximum execution time of a Job.
+
+## Motivation
+
+Extend Kueue so that kjobctl can incorporate functionalities equivalent to `-t` in [slurm](https://slurm.schedmd.com/).
+
+Different Jobs CRDs have a field with similar semantics, but there is no standard. For example `batch/Job` has `spec.activeDeadlneSeconds`, while JobSet does not have such an API for now.
+
+### Goals
+
+Introduce a Job agnostic API of setting and enforcing the maximum execution time of a Job.
+
+
+### Non-Goals
+
+Change the behavior of the Kueue's scheduler based on the maximum execution time.
+
+## Proposal
+
+A new Kueue specific label is added `kueue.x-k8s.io/max-exec-time-seconds` containing
+the maximum number of seconds the Job is expected to execute.
+
+The duration is passed to the Job's Workload.
+
+If a workload stays admitted for longer that it is expected it will be Deactivated. 
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a Batch User, I want to have the ability to set a limit on how long my job can run.
+
+### Notes/Constraints/Caveats (Optional)
+
+
+### Risks and Mitigations
+
+## Design Details
+
+### API
+
+#### Workload Spec
+
+Add a new optional filed to hold the 
+
+```go
+// WorkloadSpec defines the desired state of Workload
+// +kubebuilder:validation:XValidation:rule="has(self.priorityClassName) ? has(self.priority) : true", message="priority should not be nil when priorityClassName is set"
+type WorkloadSpec struct {
+	// ...
+
+	// MaximumExecutionTime if provided, determines the maximum time the workload can be admitted  
+	// befrore it's automatically deactivated.
+	MaximumExecutionTime *metav1.Duration `json:"maximumExecutionTime,omitempty"`
+}
+```
+
+#### Workload Status
+
+Add a optional filed to hold the 
+```go
+// WorkloadStatus defines the observed state of Workload
+type WorkloadStatus struct {
+	// ...
+
+	// AccumulatedExecutionTime holds the total duration the workload spent in Admitted state
+	// in the previous `Admit` - `Evict` cycles.
+	AccumulatedExecutionTime metav1.Duration `json:"maximumExecutionTime,omitempty"`
```

Comment:
The name of the field suggest that this is the current, often updated value. The description mentions that it is from **previous** cycles. Please consider renaming the field so that it is obvious that the field value is not current and not updated.

Reply:
done,`AccumulatedPastAdmittedTime`